op-node/rollup/driver/driverv2.go

```diff
@@ -0,0 +1,343 @@
+
+	conditions []*DriverCondition
+
+	pipeline     EnginelessPipeline
+	pipelineLock sync.Mutex // unfortunately we need a lock, since both attributes-generation and manual RPC may reset.
+
+	sequencer SequencerIface
+
+	// async gossiper for payloads to be gossiped without
+	// blocking the event loop or waiting for insertion
+	asyncGossiper      async2.AsyncGossiper
+	sequencerConductor conductor.SequencerConductor
+
+	engineController *derive.EngineController
+
+	l2 L2Chain
+
+	// TODO interop repeat-conditions: (also modify some of the above effects)
+	// TODO interop safety progression
+	// TODO interop safety reversal
+
+	payloads     *derive.PayloadsQueue
+	payloadsLock sync.RWMutex
+
+	lifetimeCtx    context.Context
+	lifetimeCancel context.CancelCauseFunc
+
+	starter sync.Once
+
+	closeWg sync.WaitGroup
+}
+
+var _ Driver = (*DriverV2)(nil)
+
+func NewDriverV2(log log.Logger,
+	sequencer SequencerIface,
+	asyncGossiper async2.AsyncGossiper,
+	sequencerConductor conductor.SequencerConductor,
+	engineController *derive.EngineController,
+	l2 L2Chain,
+) *DriverV2 {
+	lifetimeCtx, lifetimeCancel := context.WithCancelCause(context.Background())
+	d := &DriverV2{
+		log:                log,
+		lifetimeCtx:        lifetimeCtx,
+		lifetimeCancel:     lifetimeCancel,
+		payloads:           derive.NewPayloadsQueue(log, derive.MaxUnsafePayloadsMemory, derive.PayloadMemSize),
+		pipeline:           nil, // TODO
+		sequencer:          sequencer,
+		asyncGossiper:      asyncGossiper,
+		sequencerConductor: sequencerConductor,
+		engineController:   engineController,
+		l2:                 l2,
+	}
+
+	todoCondition := func() bool {
+		// TODO
+		return false
+	}
+	todoEffect := func() {
+		// TODO
+	}
+	d.attributesGeneration = d.registerCondition("attributes generation", &d.currentL1, d.checkGenerateAttributes, d.doGenerateAttributes)
+	d.unsafeBlockSyncTrigger = d.registerCondition("unsafe block sync trigger", &d.unsafeHead, d.checkUnsafeBlockSyncTrigger, d.doUnsafeBlockSyncTrigger)
+	d.unsafeBlockProcessor = d.registerCondition("unsafe block processor", &d.unsafeHead, d.checkUnsafeBlockProcessing, d.doUnsafeBlockProcessing)
+	d.sequencerAction = d.registerCondition("sequencer action", &d.unsafeHead, d.checkSequencerAction, d.doSequencerAction)
+	d.attributesForceProcessing = d.registerCondition("attributes force processing", &d.unsafeHead, todoCondition, todoEffect)
+	d.safetyProgression = d.registerCondition("safety progression", &d.safeHead, todoCondition, todoEffect)
+	d.safetyReversal = d.registerCondition("safety reversal", &d.safeHead, todoCondition, todoEffect)
+	d.finalityProgression = d.registerCondition("finality progression", &d.finalizedHead, todoCondition, todoEffect)
+	d.engineConsistency = d.registerCondition("engine consistency", &d.unsafeHead, todoCondition, todoEffect)
+	// Design note: more conditions / effects can be registered.
+	// And optionally based on hardforks, feature-flags, OP-Stack forks, etc.
```

Comment:
This is how we can make the driver more modular, to support new ways the state-transition and safety changes take effect.